buttons-examples/Web-MWC/web-button-example.md

```diff
@@ -0,0 +1,457 @@
+
+    **Example using [Material Icons](https://material.io/tools/icons/)**
+
+    ```HTML
+    <mwc-button icon="favorite" label="Button"></mwc-button>
+    <mwc-button trailingIcon="favorite" label="Button with trailing icon"></mwc-button>
+    ```
+
+    To add an icon using an `<img>` or `<svg>` tag, put the tag inside the `<mwc-button>` and add the `slot="icon"` attribute (or `slot="trailingIcon" for a trailing icon).
+
+    **Example using SVG Icons**
+
+    ```html
+    <mwc-button label="Button">
+      <svg slot="icon" aria-hidden="true" viewBox="...">
+        /*...*/
+      </svg>
+    </mwc-button>
+    ```
+
+### Making buttons accessible
+
+Material Design spec advises that touch targets should be at least 48 x 48 px.
+To meet this requirement, add the following to your button:
+
+<!-- Blocked on  https://github.com/material-components/material-components-web-components/issues/976 -->
+
+### Styling buttons with custom properties
+
+Use CSS custom properties to customize the look and feel of your buttons. 
+
+See the [Theming guide](../theming.md) for more information.
+
+## Text button
+
+[Text buttons](https://material.io/components/buttons/#text-button) are typically used for less-pronounced actions, including those located in dialogs and cards. In cards, text buttons help maintain an emphasis on card content.
+
+### Text button example
+
+API and source code:
+
+* `mwc-button`
+    * [API](https://github.com/material-components/material-components-web-components/blob/master/packages/button/README.md)
+    * [Demo](https://glitch.com/~many-witty-chihuahuas)
+    * [Source code](https://github.com/material-components/material-components-web-components/tree/master/packages/button/src)
+
+<img src="assets/web-text-button.png" alt="Text button example">
+
+```html
+<mwc-button label="Text Button"></mwc-button>
+```
+
+Text button with icon example:
+
+<img src="assets/web-text-icon-button.png" alt="Text button with bookmark icon example">
+
+```html
+<mwc-button icon="bookmark label="Text Button plus icon"></mwc-button>
+```
+
+
+### Anatomy and key properties
+
+A text button has a text label, a transparent container and an optional icon.
+
+![Text button anatomy diagram](assets/text_button_anatomy.png)
+
+1. Text label
+1. Container
+1. Icon
+
+#### Text label attributes
+
+Design Attribute | Theme value | HTML attribute | CSS property 
+---|---|---|---
+Text label | N/A | `label` | N/A 
+Text color | Primary color | N/A | None (can override `--mdc-theme-primary`)
+Typography | Default button text style | N/A | `--mdc-typography-font-family`, `--mdc-typography-button-*`
+
+For details on the available typography properties, see [Typography properties](../theming#typography-properties) in the Theming guide.
+
+#### Container attributes
+
+Design Attribute | Theme value | CSS property
+---|---|---
+Ripple color | Primary color at 16% opacity | None (can override `--mdc-theme-primary`)
```

Comment:
Override color with `--mdc-ripple-color`. Override opacity with `--mdc-ripple-press-opacity`, `--mdc-ripple-focus-opacity` and `--mdc-ripple-press-opacity`